src/lib/io/sol-iio.c

```diff
@@ -157,6 +157,8 @@ craft_filename_path(char *path, size_t size, const char *base, ...)
     len = vsnprintf(path, size, base, args);
     va_end(args);
 
+    SOL_DBG("Crafted path len: %d Buffer size: %zu", len, size);
```

Comment:
@barbieri  Do you mean `path[0] = '\0'`? If `vsnprintf` fails, I can be assured that this first char wasn't changed?